Branch-SDK/Branch-SDK/BranchUniversalObject.h

```diff
@@ -1,82 +1,235 @@
+@property (nonatomic, strong, nullable) NSString        *productBrand;
+@property (nonatomic, strong, nullable) NSString        *productCategory;
+@property (nonatomic, strong, nullable) NSString        *productVariant;
+@property (nonatomic, assign)           double          ratingAverage;
+@property (nonatomic, assign)           NSInteger       ratingCount;
+@property (nonatomic, assign)           double          ratingMaximum;
+@property (nonatomic, strong, nullable) NSString        *addressStreet;
+@property (nonatomic, strong, nullable) NSString        *addressCity;
+@property (nonatomic, strong, nullable) NSString        *addressRegion;
+@property (nonatomic, strong, nullable) NSString        *addressCountry;
+@property (nonatomic, strong, nullable) NSString        *addressPostalCode;
+@property (nonatomic, assign)           double          latitude;
+@property (nonatomic, assign)           double          longitude;
+@property (nonatomic, copy, nonnull)    NSMutableArray<NSString*> *imageCaptions;
+@property (nonatomic, copy, nonnull)    NSMutableDictionary<NSString*, NSString*> *userInfo;
+
+- (NSDictionary*_Nonnull) dictionary;
++ (BranchContentMetadata*_Nonnull) contentMetadataWithDictionary:(NSDictionary*_Nullable)dictionary;
+
+@end
+
+#pragma mark - BranchUniversalObject
+
 @interface BranchUniversalObject : NSObject
 
-@property (nonatomic, strong, nonnull) NSString *canonicalIdentifier;
+- (nonnull instancetype)initWithCanonicalIdentifier:(nonnull NSString *)canonicalIdentifier;
+- (nonnull instancetype)initWithTitle:(nonnull NSString *)title;
+
+@property (nonatomic, strong, nullable) NSString *canonicalIdentifier;
 @property (nonatomic, strong, nullable) NSString *canonicalUrl;
 @property (nonatomic, strong, nullable) NSString *title;
 @property (nonatomic, strong, nullable) NSString *contentDescription;
 @property (nonatomic, strong, nullable) NSString *imageUrl;
-// Note: properties found in metadata will overwrite properties on the BranchUniversalObject itself
-@property (nonatomic, strong, nullable) NSDictionary *metadata;
-@property (nonatomic, strong, nullable) NSString *type;
-@property (nonatomic) ContentIndexMode contentIndexMode;
-@property (nonatomic, strong, nullable) NSArray *keywords;
-@property (nonatomic, strong, nullable) NSDate *expirationDate;
-@property (nonatomic, strong, nullable) NSString *spotlightIdentifier;
-@property (nonatomic, assign) CGFloat price;
-@property (nonatomic, strong, nullable) NSString *currency;
-@property (nonatomic, assign) BOOL automaticallyListOnSpotlight;
+@property (nonatomic, strong, nullable) NSArray<NSString*> *keywords;
+@property (nonatomic, strong, nullable) NSDate   *creationDate;
+@property (nonatomic, strong, nullable) NSDate   *expirationDate;
+@property (nonatomic, assign)           BOOL      indexLocally;     //!< Index on Spotlight.
+@property (nonatomic, assign)           BOOL      indexPublicly;    //!< Index on Google, Branch, etc.
```

Comment:
can we call this locallyIndexable / publiclyIndexable for consistency?

Reply:
Done.